### DISCUSSION TITLE
child sa tunnel pinging not working

### DISCUSSION THREAD

**Discussion by joseph1267** [strongswan/strongswan#742] [Q&A] [answered]
I am using free5gc n3iwf as the server side to test my strongswan.
After the child sa exchange and and new terminal was created my topology:
(control plane)10.0.0.8 – 192.168.86.38 === 192.168.86.39 – 10.0.0.1 (ipsec tunnel)
(user plane) 60.60.0.3 – 192.168.86.38 === 192.168.86.39 – 192.168.82.5 (n3iwf gtp bind address) ------------ 127.0.0.8 (upf)
at this point, I want to ping from ue 60.60.0.3 to 192.168.82.5 but the ping is not going through.
I checked the xfrm state and policy and it had the same spi for both ways. 60.60.0.3 -> 192.168.82.5 and 192.168.82.5 --> 60.60.0.3 in free5gc, but in strongswan it had different spi for each ways. I think this is the many reason the ping is not working, but I am not sure. So, is the spi the same for the child sa for both way or different or it does not matter?

**Comment by tobiasbrunner** [comment, marked as answer]
> So, is the spi the same for the child sa for both way or different or it does not matter?

No, each host allocates an SPI for their inbound SA. Usually randomly, so they should be different (they could theoretically be the same, but both peers should show the same status output).
